From b220886cec3cee1c27f03b7c0bd3856c561d096b Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:36 -0400
Subject: [cpufreq] acpi: Use generic cpufreq routines

Message-id: <1409235879-1283-63-git-send-email-prarit@redhat.com>
Patchwork-id: 89407
O-Subject: [RHEL7.1 PATCH BZ 1134639 062/185] cpufreq: acpi: Use generic cpufreq routines
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit db9be2194a829b26d66a2a25e590d2fc39179ba3
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Thu Oct 3 20:27:56 2013 +0530

    cpufreq: acpi: Use generic cpufreq routines

    Most of the CPUFreq drivers do similar things in .exit() and .verify() routines
    and .attr. So its better if we have generic routines for them which can be used
    by cpufreq drivers then.

    This patch uses the generic verify routine in the ACPI driver.

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/acpi-cpufreq.c b/drivers/cpufreq/acpi-cpufreq.c
index b93a144..5ce7aba 100644
--- a/drivers/cpufreq/acpi-cpufreq.c
+++ b/drivers/cpufreq/acpi-cpufreq.c
@@ -516,15 +516,6 @@ out:
  return result;
 }
 
-static int acpi_cpufreq_verify(struct cpufreq_policy *policy)
-{
- struct acpi_cpufreq_data *data = per_cpu(acfreq_data, policy->cpu);
-
- pr_debug("acpi_cpufreq_verify\n");
-
- return cpufreq_frequency_table_verify(policy, data->freq_table);
-}
-
 static unsigned long
 acpi_cpufreq_guess_freq(struct acpi_cpufreq_data *data, unsigned int cpu)
 {
@@ -927,7 +918,7 @@ static struct freq_attr *acpi_cpufreq_attr[] = {
 };
 
 static struct cpufreq_driver acpi_cpufreq_driver = {
- .verify  = acpi_cpufreq_verify,
+ .verify  = cpufreq_generic_frequency_table_verify,
  .target  = acpi_cpufreq_target,
  .bios_limit = acpi_processor_get_bios_limit,
  .init  = acpi_cpufreq_cpu_init,
-- 
1.7.1